GPU/D3D11/StateMappingD3D11.cpp

```diff
@@ -293,6 +293,34 @@ void DrawEngineD3D11::ApplyDrawState(int prim) {
 				keys_.depthStencil.stencilWriteMask = stencilState.writeMask;
 				dynState_.useStencil = true;
 				dynState_.stencilRef = stencilState.testRef;
+
+				// Nasty special case for Spongebob and similar where it tries to write zeros to alpha/stencil during
+				// depth-fail. We can't write to alpha then because the pixel is killed. However, we can invert the depth
+				// test and modify the alpha function...
+				if (gstate.isDepthTestEnabled() && !gstate.isDepthWriteEnabled() && stencilState.zFail == GE_STENCILOP_ZERO &&
+					stencilState.sFail == GE_STENCILOP_KEEP && stencilState.zPass == GE_STENCILOP_KEEP &&
+					stencilState.testFunc == GE_COMP_ALWAYS &&
+					stencilState.writeMask == 0xFF && stencilState.testMask == 0xFF && stencilState.testRef == 0xFF) {
```

Comment:
testMask/testRef ought to be irrelevant if it's ALWAYS.  But don't we need to verify the actual depth test itself?  What about if it's writing color?  How can we safely inverse the depth test without caring about those?

-[Unknown]

Reply:
You're right, I got way too excited about it working and forgot to finish the checks. Will do.

Reply:
testMask is irrelevant yes but testRef gets written - although you're right, with KEEP/KEEP/ZERO nothing is reading it.

I'll remove the checks for those too.